### DISCUSSION TITLE
API Support for frontend frameworks

### DISCUSSION THREAD

**Discussion by Thinkro** [spatie/laravel-medialibrary#3176] [Ideas]
I have recently been testing the package and purchased the media library pro version. However, I think it would be beneficial if the library had better support for APIs and frontend frameworks like Vue, React and Svelte etc.

I have noticed that there are some discussions and resources available in the repo on how people are solving the problem of working with APIs and frontend frameworks, but it’s mostly doing things to work around the package, maybe there’s way this could be streamlined?

The possibility to create something like:
```
<picture>
  <source
    width="600" height="480"
    type="image/avif"
    loading="lazy"
    srcset="
      /image/image_600.jpg   600w,
      /image/image_800.jpg   800w,
      /image/image_1400.jpg 1400w,
    "
  />
  <source
    width="600" height="480"
    type="image/webp"
    loading="lazy"
    srcset="
      /image/image_600.jpg   600w,
      /image/image_800.jpg   800w,
      /image/image_1400.jpg 1400w,
    "
  />
  <img
    width="600" height="480"
    loading="lazy"
    srcset="
      /image/image_600.jpg   600w,
      /image/image_800.jpg   800w,
      /image/image_1400.jpg 1400w,
    "
    src="image/image_800.jpg"
    alt="A dog chasing a ball."
  />
</ picture>

```
Having the ability to pass the height and width for images is important for automatic aspect ratio calculation to avoid content shift.
https://chromestatus.com/feature/5737185317748736

I would appreciate your consideration of this request and I look forward to seeing the library improved with these features in the future. If anyone has solved this or have suggestions on how to implement this using the current available methods please let me know.

Thank you!

Some of the issues/discussions mentioning API/JSON endpoints:
https://github.com/spatie/laravel-medialibrary/issues/892
https://github.com/spatie/laravel-medialibrary/issues/1100
https://github.com/spatie/laravel-medialibrary/issues/1302
https://github.com/spatie/laravel-medialibrary/issues/1262
https://github.com/spatie/laravel-medialibrary/discussions/2573
https://github.com/spatie/laravel-medialibrary/issues/1135  

I’ve gathered some resources from the repo on how people are solving/hacking around this:
https://github.com/spatie/laravel-medialibrary/issues/892
https://gist.github.com/adiachenko/508d227ce3ce444aa1e649965a7f157a

**Comment by websitevirtuoso** [comment]
Thanks got info. yes, I think ewe have to implement it manually. each use case individual
